update_accounts.py

```diff
@@ -30,42 +46,45 @@ def refresh_token(api_url: str, refresh_token: str, client_id: str, client_secre
         log.error("Could not refresh access token")
         return ""
 
-def update_discovery_url(api_url: str, discovery_url: str):
-    with open(args.canvas_token) as token_file:
-        TOKEN = json.load(token_file)
-    with open(args.oauth_settings) as oauth_file:
-        OAUTH = json.load(oauth_file)
-
-    access_token = TOKEN.get("access_token")
+def update_discovery_url(canvas_url: str, discovery_url: str, token: Token, settings: OAuthSettings):
+    log.info(f"Updating discovery url on {canvas_url} to '{discovery_url}'")
+    access_token = token.access_token
     try:
-        CANVAS = Canvas(api_url, access_token)
-        account = CANVAS.get_account(1)
+        canvas = Canvas(canvas_url, access_token)
+        account = canvas.get_account(1)
     except InvalidAccessToken:
         log.info("Access token is invalid, attempting to refresh!")
-        access_token = refresh_token(api_url, TOKEN.get("refresh_token"),
-                                     OAUTH.get("client_id"), OAUTH.get("client_secret"), OAUTH.get("redirect_uri"))
-        CANVAS = Canvas(api_url, access_token)
-        account = CANVAS.get_account(1)
+        access_token = refresh_token(canvas_url, token.refresh_token,
+                                     settings.client_id, settings.client_secret, discovery_url)
+        log.debug(access_token)
+        canvas = Canvas(canvas_url, access_token)
+        account = canvas.get_account(settings.account_id)
 
+    try:
         # TODO: Make this URL configurable
         account.update_account_auth_settings(
             sso_settings={"auth_discovery_url": discovery_url})
+    except Exception as e:
+        log.error(f"Error updating {api_url} {str(e)}")
 
-parser = argparse.ArgumentParser(description='Update Canvas Test/Beta Settings')
-parser.add_argument('canvas_token', type=str, help='Canvas Token File')
-parser.add_argument('oauth_settings', type=str, help='OAuth Settings File')
-
-args = parser.parse_args()
+def main():
+    parser = argparse.ArgumentParser(description='Update Canvas Test/Beta Settings')
+    parser.add_argument('canvas_token', type=str, help='Canvas Token File')
+    parser.add_argument('settings', type=str, help='Settings File')
 
-if not os.path.isfile(args.canvas_token):
-    log.warn("Canvas Token file does not exist, must provide the path to this file")
-    sys.exit()
-if not os.path.isfile(args.oauth_settings):
-    log.warn("OAuth Settings file does not exist, must provide the path to this file")
-    sys.exit()
+    args = parser.parse_args()
 
-# TODO: Make this URL configurable?
-update_discovery_url("https://umich.beta.instructure.com", "")
-update_discovery_url("https://canvas-test.it.umich.edu", "https://canvas-dev.dsc.umich.edu/")
+    try:
+        with open(args.canvas_token) as token_file:
+            token = Token(**json.load(token_file))
+        with open(args.settings) as settings_file:
```

Comment:
Both json config files are mentioned n the README, but only one sample file - oauth_settings_sample.json is provided in code base. I wonder whether the sample file of canvas_token.json should also be added?

Reply:
A sample of what this file looks like is in the README and the file is generated after running the script in util. There's nothing that can be manually entered into that file.

If you wanted to test this you can get the values from Openshift secrets or go through the process to generate this json file.

Reply:
I'd thought about making the process to get this more automated. But often this requires 2 people, one to run the script here and another to login as an admin user. adding more automation for a seldom used process didn't seem worthwhile.

Maybe in the future the process for generating these tokens will be easier.